tests/test_funcs.py

```diff
@@ -136,6 +136,7 @@ def test_device(self, xp: ModuleType, device: Device):
         x = xp.asarray([1, 2, 3], device=device)
         assert get_device(cov(x)) == device
 
+    @pytest.mark.skip_xp_backend(Backend.NUMPY_READONLY)
```

Comment:
Notably, read-only numpy disallows `__iadd__` etc., whereas jax and sparse simply don't define these methods, which causes them to fall back to `a = a.__add__(b)`